src/test/java/org/folio/rest/api/CallNumberNormalizedTestBase.java

```diff
@@ -0,0 +1,151 @@
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+
+import java.util.List;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+
+@RunWith(JUnitParamsRunner.class)
+public abstract class CallNumberNormalizedTestBase extends TestBaseWithInventoryUtil {
+
+  protected abstract List<String> searchByCallNumberNormalized(String callNumber) throws Exception;
+
+  @Parameters({
+    "germ 350",
+    "germ350",
+    "Germ 350/"
+  })
+  @Test
+  public void appliesRightTruncationToSearchQuery(String searchQuery) throws Exception {
+    final List<String> callNumbersFound = searchByCallNumberNormalized(searchQuery);
+
+    assertThat(callNumbersFound, hasSize(3));
+    assertThat(callNumbersFound, allOf(
+      hasItem("H Germ 350/35: 1"),
+      hasItem("H Germ 350/35: 2"),
+      hasItem("F Germ 350/1")
+    ));
+  }
+
+  @Test
+  public void canApplyRightTruncationForSuffix() throws Exception {
+    final List<String> callNumbersFound = searchByCallNumberNormalized("GE77 .F73 2014 Curriculum");
+
+    assertThat(callNumbersFound, hasSize(1));
+    assertThat(callNumbersFound, hasItem("GE77 .F73 2014 Curriculum Materials Collection"));
+  }
+
+  @Test
+  public void canIgnoreNonAlphanumericCharacters() throws Exception {
+    final List<String> callNumbersFound = searchByCallNumberNormalized("12");
+
+    assertThat(callNumbersFound, hasSize(4));
+    assertThat(callNumbersFound, allOf(
+      hasItem("1—2"),
+      hasItem("1 . 2"),
+      hasItem("1 . . 2"),
+      hasItem("1— —2")
+    ));
+  }
+
+  @Test
+  public void canSkipPrefix() throws Exception {
+    final List<String> callNumbersFound = searchByCallNumberNormalized("ABC123.1 R15 2018");
+
+    assertThat(callNumbersFound, hasSize(1));
+    assertThat(callNumbersFound, hasItem("Oversize ABC123.1 .R15 2018"));
+  }
+
+  @Test
+  public void canSkipNonAlphanumericCharactersInPrefix() throws Exception {
+    final List<String> callNumbersFound = searchByCallNumberNormalized("Oversize BX1935.A23 1959");
+
+    assertThat(callNumbersFound, hasSize(1));
+    assertThat(callNumbersFound, hasItem("++ Oversize BX1935 .A23 1959"));
+  }
+
+  @Test
+  public void canSearchByCallNumberAndIgnorePrefixAndSuffix() throws Exception {
+    final List<String> callNumbersFound = searchByCallNumberNormalized("S537.N56 C82");
+
+    assertThat(callNumbersFound, hasSize(2));
+    assertThat(callNumbersFound, allOf(
+      hasItem("Rare Books S537.N56 C82 ++"),
+      hasItem("Rare Books S537.N56 C82 Foo")
+    ));
+  }
+
+  @Test
+  public void cannotSearchByNonAlphanumericsOnly() throws Exception {
+    assertThat(searchByCallNumberNormalized("++"), hasSize(0));
+  }
+
+  @Parameters({
+    "Rare Books S537 N56 C82",
+    "Rare Books S537 N56",
+    "rarebookss537n56c82",
+  })
+  @Test
+  public void canSearchByPrefixAndCallNumber(String searchQuery) throws Exception {
```

Comment:
How is this different to the first test?

Reply:
In this case call numbers also have a suffix.

Reply:
Can the tests be renamed to reflect that difference? For example, CanSearchByPrefixAndCallNumberEvenWhenMatchesHaveSuffixes (I'm not super keen on this name, hopefully you can this of a better one)

Reply:
Renamed.